StorySquad/SubmissionsAPI.swift

```diff
@@ -0,0 +1,429 @@
+            }
+            
+        }.resume()
+    }
+    
+    //MARK: - Attempts to upload a drawing for the submission with the given ID.
+    func submitDrawing(child: ChildSubmission, draw: UIImage, completion: @escaping(Result<String, NetworkingError>) -> Void) {
+        
+        guard let childBearer = childBearer else {
+            completion(Result.failure(NetworkingError.noBearer))
+            return
+        }
+        
+        let registerURL = baseURL
+            .appendingPathComponent("/submit/draw/\(child.id)")
+        
+        let boundary = UUID().uuidString
+        
+        var request = URLRequest(url: registerURL)
+        request.httpMethod = HTTPMethod.post.rawValue
+        request.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        request.setValue("\(childBearer.token)", forHTTPHeaderField: HeaderNames.authorization.rawValue)
+        
+        let httpBody = NSMutableData()
+        if let imageData = draw.jpegData(compressionQuality: 1) {
+            httpBody.append(convertFileData(mimeType: "image/jpeg",
+                                            fileData: imageData,
+                                            using: boundary))
+        }
+        
+        httpBody.appendString("--\(boundary)--")
+        
+        request.httpBody = httpBody as Data
+        
+        print(String(data: httpBody as Data, encoding: .utf8)!)
+        
+        URLSession.shared.dataTask(with: request) { (data, response, error) in
+            
+            if let error = error {
+                print("Error getting response: \(error)")
+                completion(.failure(.serverError(error)))
+                return
+            }
+            
+            if let response = response as? HTTPURLResponse {
+                if response.statusCode == 201 {
+                    print("Good response")
+                    completion(.success("Image submitted"))
+                    return
+                } else {
+                    print("Bad response, code: \(response.statusCode)")
+                    completion(.failure(.unexpectedStatusCode(response.statusCode)))
+                    return
+                }
+            }
+            
+            guard let data = data else {
+                completion(.failure(.noData))
+                return
+            }
+            
+            let jsonData = try? JSONSerialization.jsonObject(with: data, options: .allowFragments)
+            if let json = jsonData as? [String: Any] {
+                print(json)
+            }
+            
+            do {
+                let image = try JSONDecoder().decode(Image.self, from: data)
+                print(image)
+                // TODO: Do something with decoded object
+            } catch {
+                NSLog("Error decoding image: \(error)")
+                completion(.failure(.badDecode))
+            }
+            
+        }.resume()
+    }
+    
```

Comment:
This needs to access the current user's id instead of passing in a submission, and it should return an array [Submission]. In the "do" block, make sure you are decoding the data as an array, and remember to call completion.